src/ziggurat/producer.clj

```diff
@@ -41,109 +41,38 @@
-                                   (s/optional-key :metric-num-samples)                    s/Any
-                                   (s/optional-key :transaction-timeout)                   s/Any
-                                   (s/optional-key :retries)                               s/Any
-                                   (s/optional-key :retry-backoff-ms)                      s/Any
-                                   (s/optional-key :receive-buffer)                        s/Any
-                                   (s/optional-key :partitioner-class)                     s/Any
-                                   (s/optional-key :max-block-ms)                          s/Any
-                                   (s/optional-key :metrics-reporter-classes)              s/Any
-                                   (s/optional-key :compression-type)                      s/Any
-                                   (s/optional-key :max-request-size)                      s/Any
-                                   (s/optional-key :delivery-timeout-ms)                   s/Any
-                                   (s/optional-key :metrics-sample-window-ms)              s/Any
-                                   (s/optional-key :request-timeout-ms)                    s/Any
-                                   (s/optional-key :buffer-memory)                         s/Any
-                                   (s/optional-key :interceptor-classes)                   s/Any
-                                   (s/optional-key :linger-ms)                             s/Any
-                                   (s/optional-key :connections-max-idle-ms)               s/Any
-                                   (s/optional-key :acks)                                  s/Any
-                                   (s/optional-key :enable-idempotence)                    s/Any
-                                   (s/optional-key :metrics-recording-level)               s/Any
-                                   (s/optional-key :transactional-id)                      s/Any
-                                   (s/optional-key :reconnect-backoff-max-ms)              s/Any
-                                   (s/optional-key :client-dns-lookup)                     s/Any
-                                   (s/optional-key :max-in-flight-requests-per-connection) s/Any
-                                   (s/optional-key :send-buffer)                           s/Any
-                                   (s/optional-key :batch-size)                            s/Any})
-
-(def valid-configs? (partial s/validate ProducerConfigSchema))
-
-(def explain-str (partial s/explain ProducerConfigSchema))
-
-(defn property->fn [field-name]
-  (let [raw-field-name (condp = field-name
-                         :max-in-flight-requests-per-connection "org.apache.kafka.clients.producer.ProducerConfig/%s"
-                         :retries-config "org.apache.kafka.clients.producer.ProducerConfig/RETRIES_CONFIG"
```

Comment:
Following configs for producer namespace should be there in mapping-table for backward compatibility reasons.

:retries-config
:key-serializer-class
:value-serializer-class

Reply:
already added `:retries-config`, others are already there,,,